Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -255,24 +300,94 @@ struct NSFileManagerMoveOptions: ExpressibleByArrayLiteral {
+                    throw error
+                }
+                return
+            }
+            if faAttributes.dwFileAttributes & FILE_ATTRIBUTE_READONLY == FILE_ATTRIBUTE_READONLY {
+                guard SetFileAttributesW($0, faAttributes.dwFileAttributes & ~FILE_ATTRIBUTE_READONLY) else {
+                    throw CocoaError.removeFileError(GetLastError(), path)
+                }
+            }
+            if faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == 0 || faAttributes.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT == FILE_ATTRIBUTE_REPARSE_POINT {
+                if faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY {
+                    guard RemoveDirectoryW($0) else {
+                        throw CocoaError.removeFileError(GetLastError(), path)
+                    }
+                } else {
+                    guard DeleteFileW($0) else {
+                        throw CocoaError.removeFileError(GetLastError(), path)
+                    }
+                }
+            }
+        }
+
+        var stack = [path]
+        while let directory = stack.popLast() {
+            guard filemanager?._shouldRemoveItemAtPath(directory) ?? true else { continue }
+            try directory.withNTPathRepresentation {
+                if RemoveDirectoryW($0) { return }
+                let dwError: DWORD = GetLastError()
+                guard dwError == ERROR_DIR_NOT_EMPTY else {
+                    throw CocoaError.removeFileError(dwError, directory)
+                }
+                stack.append(directory)
+
+                for entry in _Win32DirectoryContentsSequence(path: directory, appendSlashForDirectory: false, prefix: [directory]) {
+                    try entry.fileName.withNTPathRepresentation {
+                        if entry.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY,
+                                entry.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT != FILE_ATTRIBUTE_REPARSE_POINT {
+                            stack.append(entry.fileNameWithPrefix)
+                        } else {
+                            if entry.dwFileAttributes & FILE_ATTRIBUTE_READONLY == FILE_ATTRIBUTE_READONLY {
+                                guard SetFileAttributesW($0, entry.dwFileAttributes & ~FILE_ATTRIBUTE_READONLY) else {
+                                    throw CocoaError.removeFileError(GetLastError(), path)
+                                }
+                            }
+                            guard filemanager?._shouldRemoveItemAtPath(entry.fileNameWithPrefix) ?? true else { return }
+                            if entry.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY {
+                                guard RemoveDirectoryW($0) else {
+                                    throw CocoaError.removeFileError(GetLastError(), entry.fileName)
```

Comment:
Should we check with the delegate here as well to determine if we should continue?